ui/src/components/param-viewer.tsx

```diff
@@ -1,26 +1,46 @@
+import { BlankParam } from './blank-param';
 import { Param } from './param';
 import { ScrollArea } from './ui/scroll-area';
-import { Parameter } from '@/types/parameter';
+import { Parameter, ParameterType } from '@/types/parameter';
 
 interface ParamViewerProps {
   parameters: Parameter[];
   generatedValues: string[];
+  blankParamValues: string[];
+  setBlankParam: (index: number, value: string) => void;
 }
 
-export function ParamViewer({ parameters, generatedValues }: ParamViewerProps) {
+export function ParamViewer({
+  parameters,
+  generatedValues,
+  blankParamValues,
+  setBlankParam,
+}: ParamViewerProps) {
   return (
     <ScrollArea id="parameters" className="h-40 rounded-md border">
       <div className="p-4">
-        {parameters.map((parameter, index) => (
-          <Param
-            key={index}
-            parameter={parameter}
-            generatedValue={
-              index < generatedValues.length ? generatedValues[index] : ''
-            }
-            index={index}
-          />
-        ))}
+        {parameters.map((parameter, index) => {
+          if (parameter.type == ParameterType.Blank) {
+            let blankIndex = parameters.filter((p, i) => p.type == ParameterType.Blank && i < index).length;
```

Comment:
super nit: something like
```suggestion
            let blankIndex = parameters
              .slice(0, index)
              .filter((p) => p.type == ParameterType.Blank).length;
```
would be more readable (ignore if u disagree)